### ISSUE TITLE
API requests with long query parameters do not return a response

### ISSUE THREAD

**Issue by mdeboer** [LearningLocker/learninglocker#1361]
#### What version were you using?
v3.8.1 and v2.5.0 (xAPI) - open source

#### What steps can we follow to reproduce the behaviour?
1. Create lot of organisations
2. Assign all those organisations to one user
3. Log in as that user through the UI

#### What is the actual behaviour?
The `GET /connection/organisation?filter=...` request to fetch organisations fails immediately with no status code. The console logs that popsicle throws an error with code `EUNAVAILABLE`.

I copied the request and executed it in [Paw](https://paw.cloud/) to get the same result. Removing the filter parameter shows a correct response.

#### What is the expected behavior?
A normal response.

#### Is there any additional information that will help us replicate/understand the problem?
I thought it had to do with the `HTTP_MAX_HEADER_SIZE` reset to 8kb on NodeJS v8.14.0 but using v8.15.0 or v8.16.0 with the newly added `--max-http-header-size` set to 80kb or more didn't fix it so far when using PM2 with `node_args` set. Still have to try without PM2.

**Comment by ryansmith94** [comment]
@Ian247 did you say you'd seen this when we spoke yesterday?

**Comment by ryansmith94** [comment]
Ok yeah looks like an issue we've seen before, I'll set a reminder to discuss this internally this week.

**Comment by Ian247** [comment]
Yes, I have experienced the same symptoms as a result of being assigned to a lot of orgs.

**Comment by ht2** [comment]
Pretty sure this is to do with nginx's [large_client_header_buffers](http://nginx.org/en/docs/http/ngx_http_core_module.html#large_client_header_buffers) value being too small. 

We do actually make this bigger here: https://github.com/LearningLocker/learninglocker/blob/master/nginx.conf.example#L11, but perhaps it's still not big enough?

@mdeboer Could you confirm the value of 'large_client_header_buffers' in your nginx config file (should be in `/etc/nginx/conf.d/`)

**Comment by mdeboer** [comment]
Thanks, I'll look into it!

**Comment by mdeboer** [comment]
Okay I fixed it and it was a combination of Nginx with NodeJS (PM2).

To fix it I simply added the `--max-http-header-size=16384` (16kb) option to the [PM2 configuration](https://futurestud.io/tutorials/pm2-how-to-start-your-app-with-node-js-v8-arguments). Then I had to increase the proxy buffer size for Nginx to 16kb as well.

I figured that would do the trick but I still got errors, upon setting the nginx error_log level to debug I noticed that there is a specific parameter when using HTTP2 called `http2_max_field_size` which by default is set to 4kb. I increased this to 16kb and it all works like a charm now.

Maybe add this to the documentation?

Thanks all for your help 👍
